File: tests/test_autobench.py
# -*- coding: utf-8 -*-
from __future__ import print_function

from bairstow.autocorr import initial_autocorr, pbairstow_autocorr
from bairstow.rootfinding import initial_guess, pbairstow_even


def run_autocorr():
    """[summary]

    Returns:
        [type]: [description]
    """
    p = [10.0, 34.0, 75.0, 94.0, 150.0, 94.0, 75.0, 34.0, 10.0]
    vr0s = initial_autocorr(p)
    _, niter, _ = pbairstow_autocorr(p, vr0s)
    return niter


def run_pbairstow():
    """[summary]

    Returns:
        [type]: [description]
    """
    p = [10.0, 34.0, 75.0, 94.0, 150.0, 94.0, 75.0, 34.0, 10.0]
    vr0s = initial_guess(p)
    _, niter, _ = pbairstow_even(p, vr0s)
    return niter


def test_autocorr(benchmark):
    """[summary]

    Arguments:
        benchmark ([type]): [description]
    """
    result = benchmark(run_autocorr)
    assert result <= 12


def test_pbairstow(benchmark):
    """[summary]

    Arguments:
        benchmark ([type]): [description]
    """
    result = benchmark(run_pbairstow)
    assert result <= 11

File: tests-pyrs/test_autobench.rs
use std::collections::HashMap;
use std::*;

use __future__::print_function;
use bairstow::autocorr::{initial_autocorr, pbairstow_autocorr};
use bairstow::rootfinding::{initial_guess, pbairstow_even};
fn run_autocorr<RT>() -> RT {
    "[summary]

    Returns:
        [type]: [description]
    ";
    let p = vec![10.0, 34.0, 75.0, 94.0, 150.0, 94.0, 75.0, 34.0, 10.0];
    let vr0s = initial_autocorr(p);
    let (_, niter, _) = pbairstow_autocorr(p, vr0s);
    return niter;
}
fn run_pbairstow<RT>() -> RT {
    "[summary]

    Returns:
        [type]: [description]
    ";
    let p = vec![10.0, 34.0, 75.0, 94.0, 150.0, 94.0, 75.0, 34.0, 10.0];
    let vr0s = initial_guess(p);
    let (_, niter, _) = pbairstow_even(p, vr0s);
    return niter;
}
fn test_autocorr<T0>(benchmark: T0) {
    "[summary]

    Arguments:
        benchmark ([type]): [description]
    ";
    let result = benchmark(run_autocorr);
    assert!(result <= 13);
}
fn test_pbairstow<T0>(benchmark: T0) {
    "[summary]

    Arguments:
        benchmark ([type]): [description]
    ";
    let result = benchmark(run_pbairstow);
    assert!(result <= 12);
}
